File: src/ProjectR/content/scripts/spells/ExplosiveFist.lua
function GetTargetType()
    return Decaying
end

function GetName()
  return "Explosive Fist"
end

function GetDescription()
  return "Attack your target with an arcane-enchanted punch.\nEnemies in proximity also take damage."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .88
end

function GetDelay()
  return .3
end

function GetSpellType()
  return Composite
end

function GetMasteries()
  return { ARC }
end

function SpellEffect()
  local damage = ((3.75 * aAD + 3.75 * aMD) * (aARC/100) - (1.25 * dDEF + 1.25 * dMR)) * (100/dARC) / modifier
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/ExplosiveFist.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class ExplosiveFist : SpellScriptBase
    {
        public override string Name { get { return "Explosive Fist"; } }
        public override string Description { get { return "Attack your target with an arcane-enchanted punch.\nEnemies in proximity also take damage."; } }

        public override TargetType TargetType { get { return TargetType.Decaying; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.ARC, }; } }
        public override SpellType SpellType { get { return SpellType.Composite; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 88; } }
        public override double Delay { get { return .3; } }

        public override void SpellEffect(ICharacter caster, ICharacter target, double modifier)
        {
            var damage = ((3.75 * cAD + 3.75 * cMD) * (cARC / 100) - (1.25 * tDEF + 1.25 * tMR)) * (100 / tARC) / modifier;
            DealDamage(damage);
        }
    }
}